tests/helpers/test_config_validation.py

```diff
@@ -56,7 +58,67 @@ def test_port():
             schema(value)
 
     for value in ('1000', 21, 24574):
-        schema(value)
+        assert schema(value)
+
+
+def test_isfile():
+    """Validate that the value is an existing file."""
+    schema = vol.Schema(cv.isfile)
+
+    for value in ('invalid', None, -1, 0, 80000, '/fhadh/djd/test.txt'):
+        with pytest.raises(vol.Invalid):
+            schema(value)
+
+    with tempfile.TemporaryDirectory() as tmp_path:
+        tmp_file = os.path.join(tmp_path, "test.txt")
+        with open(tmp_file, "w") as tmp_handl:
+            tmp_handl.write("Test File")
+        assert schema(tmp_file)
+
+
+def test_filename():
+    """Validate if file is writeable and can be created."""
+    schema = vol.Schema(cv.filename)
+
+    for value in (None, '/fhadh/djd/test.txt', os.getcwd()):
+        with pytest.raises(vol.Invalid):
+            schema(value)
+
+    with tempfile.TemporaryDirectory() as tmp_path:
+        tmp_file = os.path.join(tmp_path, "test.txt")
+        with open(tmp_file, "w") as tmp_handl:
+            tmp_handl.write("Test File")
+        assert schema(tmp_file)
+
+    assert schema(os.path.join(os.path.expanduser("~"), 'test.txt'))
+    assert schema("test.txt")
+
+
+def test_isdir():
+    """Validate that the value is an existing folder."""
+    schema = vol.Schema(cv.isdir)
+
+    for value in ('invalid', None, -1, 0, 80000, '/fhadh/djd'):
+        with pytest.raises(vol.Invalid):
+            schema(value)
+
+    with tempfile.TemporaryDirectory() as tmp_path:
+        assert schema(tmp_path)
+
+
+def test_dirname():
+    """Validate if folder is writeable."""
+    schema = vol.Schema(cv.dirname)
+
+    for value in (None, '/fhadh/djd'):
```

Comment:
Better to have a temp dir be removed and then use that path then to just type a random path